*** evcc/templates/definition/charger/ocpp.yaml 2024-08-21 07:11:58.692632391 +0800
--- evcc/templates/definition/charger/ocpp.yaml.new     2024-08-21 07:10:35.298927896 +0800
***************
*** 47,53 ****
        * Local network connection

        The specific configuration and the actual usable functionality depend on the charger model and its software.
!   evcc: ["sponsorship", "skiptest"]
  params:
    - preset: ocpp
    - name: autostart
--- 47,53 ----
        * Local network connection

        The specific configuration and the actual usable functionality depend on the charger model and its software.
! #  evcc: ["sponsorship", "skiptest"]
  params:
    - preset: ocpp
    - name: autostart
*** evcc/templates/definition/charger/ocpp-pulsarplus.yaml      2024-08-21 07:11:45.964372238 +0800
--- evcc/templates/definition/charger/ocpp-pulsarplus.yaml.new  2024-08-21 07:10:48.775203340 +0800
***************
*** 19,25 ****
        * URL: ws://[evcc-adresse]:8887/ (local network connection)
        * Charge Point Identity: Custom value (e.g. serial number of charger) which is reused in configuration as *stationid*
        * Password: leave empty
!   evcc: ["sponsorship", "skiptest"]
  params:
    - preset: ocpp
  render: |
--- 19,25 ----
        * URL: ws://[evcc-adresse]:8887/ (local network connection)
        * Charge Point Identity: Custom value (e.g. serial number of charger) which is reused in configuration as *stationid*
        * Password: leave empty
! #  evcc: ["sponsorship", "skiptest"]
  params:
    - preset: ocpp
  render: |
*** evcc/charger/ocpp.go        2024-10-25 21:14:49.229049614 +0800
--- evcc/charger/ocpp.go.new    2024-10-26 06:37:36.858803828 +0800
***************
*** 28,34 ****
        "github.com/evcc-io/evcc/charger/ocpp"
        "github.com/evcc-io/evcc/core/loadpoint"
        "github.com/evcc-io/evcc/util"
!       "github.com/evcc-io/evcc/util/sponsor"
        "github.com/lorenzodonini/ocpp-go/ocpp1.6/core"
        "github.com/lorenzodonini/ocpp-go/ocpp1.6/types"
        "github.com/samber/lo"
--- 28,34 ----
        "github.com/evcc-io/evcc/charger/ocpp"
        "github.com/evcc-io/evcc/core/loadpoint"
        "github.com/evcc-io/evcc/util"
! //    "github.com/evcc-io/evcc/util/sponsor"
        "github.com/lorenzodonini/ocpp-go/ocpp1.6/core"
        "github.com/lorenzodonini/ocpp-go/ocpp1.6/types"
        "github.com/samber/lo"
***************
*** 92,100 ****
                return c, err
        }

!       if !sponsor.IsAuthorized() {
!               return nil, api.ErrSponsorRequired
!       }

        var (
                powerG, totalEnergyG, socG func() (float64, error)
--- 92,100 ----
                return c, err
        }

! //    if !sponsor.IsAuthorized() {
! //            return nil, api.ErrSponsorRequired
! //    }

        var (
                powerG, totalEnergyG, socG func() (float64, error)
