*** evcc/vehicle/tesla.go	2024-02-24 08:34:28.285811074 +0800
--- evcc/vehicle/tesla.go.new	2024-02-24 08:36:39.811461111 +0800
***************
*** 8,14 ****
  	"github.com/evcc-io/evcc/api"
  	"github.com/evcc-io/evcc/util"
  	"github.com/evcc-io/evcc/util/request"
! 	"github.com/evcc-io/evcc/util/sponsor"
  	"github.com/evcc-io/evcc/util/transport"
  	"github.com/evcc-io/evcc/vehicle/tesla"
  	teslaclient "github.com/evcc-io/tesla-proxy-client"
--- 8,14 ----
  	"github.com/evcc-io/evcc/api"
  	"github.com/evcc-io/evcc/util"
  	"github.com/evcc-io/evcc/util/request"
! //	"github.com/evcc-io/evcc/util/sponsor"
  	"github.com/evcc-io/evcc/util/transport"
  	"github.com/evcc-io/evcc/vehicle/tesla"
  	teslaclient "github.com/evcc-io/tesla-proxy-client"
***************
*** 98,104 ****
  	pc := request.NewClient(log)
  	pc.Transport = &transport.Decorator{
  		Decorator: transport.DecorateHeaders(map[string]string{
! 			"X-Auth-Token": sponsor.Token,
  		}),
  		Base: hc.Transport,
  	}
--- 98,104 ----
  	pc := request.NewClient(log)
  	pc.Transport = &transport.Decorator{
  		Decorator: transport.DecorateHeaders(map[string]string{
! 			"X-Auth-Token": "gibberish",
  		}),
  		Base: hc.Transport,
  	}
*** evcc/vehicle/tesla/controller.go    2024-04-07 10:07:05.283869937 +0800
--- evcc/vehicle/tesla/controller.go.new        2024-04-07 10:09:18.578318067 +0800
***************
*** 7,17 ****
        "slices"

        "github.com/evcc-io/evcc/api"
!       "github.com/evcc-io/evcc/util/sponsor"
        "github.com/evcc-io/tesla-proxy-client"
  )

! const ProxyBaseUrl = "https://tesla.evcc.io"

  type Controller struct {
        vehicle *tesla.Vehicle
--- 7,17 ----
        "slices"

        "github.com/evcc-io/evcc/api"
! //    "github.com/evcc-io/evcc/util/sponsor"
        "github.com/evcc-io/tesla-proxy-client"
  )

! const ProxyBaseUrl = "https://tesla.haveacry.com"

  type Controller struct {
        vehicle *tesla.Vehicle
***************
*** 41,49 ****

  // MaxCurrent implements the api.CurrentController interface
  func (v *Controller) MaxCurrent(current int64) error {
!       if !sponsor.IsAuthorized() {
!               return api.ErrSponsorRequired
!       }

        v.current = current
        v.dataG.Reset()
--- 41,49 ----

  // MaxCurrent implements the api.CurrentController interface
  func (v *Controller) MaxCurrent(current int64) error {
! //    if !sponsor.IsAuthorized() {
! //            return api.ErrSponsorRequired
! //    }

        v.current = current
        v.dataG.Reset()
***************
*** 62,70 ****

  // ChargeEnable implements the api.ChargeController interface
  func (v *Controller) ChargeEnable(enable bool) error {
!       if !sponsor.IsAuthorized() {
!               return api.ErrSponsorRequired
!       }

        var err error

--- 62,70 ----

  // ChargeEnable implements the api.ChargeController interface
  func (v *Controller) ChargeEnable(enable bool) error {
! //    if !sponsor.IsAuthorized() {
! //            return api.ErrSponsorRequired
! //    }

        var err error

*** evcc/templates/definition/charger/ocpp.yaml 2024-08-21 07:11:58.692632391 +0800
--- evcc/templates/definition/charger/ocpp.yaml.new     2024-08-21 07:10:35.298927896 +0800
***************
*** 47,53 ****
        * Local network connection

        The specific configuration and the actual usable functionality depend on the charger model and its software.
!   evcc: ["sponsorship"]
  params:
    - preset: ocpp
    - name: autostart
--- 47,53 ----
        * Local network connection

        The specific configuration and the actual usable functionality depend on the charger model and its software.
! #  evcc: ["sponsorship"]
  params:
    - preset: ocpp
    - name: autostart
*** evcc/templates/definition/charger/ocpp-pulsarplus.yaml      2024-08-21 07:11:45.964372238 +0800
--- evcc/templates/definition/charger/ocpp-pulsarplus.yaml.new  2024-08-21 07:10:48.775203340 +0800
***************
*** 19,25 ****
        * URL: ws://[evcc-adresse]:8887/ (local network connection)
        * Charge Point Identity: Custom value (e.g. serial number of charger) which is reused in configuration as *stationid*
        * Password: leave empty
!   evcc: ["sponsorship"]
  params:
    - preset: ocpp
  render: |
--- 19,25 ----
        * URL: ws://[evcc-adresse]:8887/ (local network connection)
        * Charge Point Identity: Custom value (e.g. serial number of charger) which is reused in configuration as *stationid*
        * Password: leave empty
! #  evcc: ["sponsorship"]
  params:
    - preset: ocpp
  render: |
